workspaces/local-cli/src/shared/analytics.ts

```diff
@@ -37,36 +42,51 @@ export async function getUser(): Promise<IUser | null> {
   });
 }
 
-export async function track(event: string, properties: any = {}) {
-  await new Promise((resolve, reject) => {
-    getUser().then((user) => {
-      if (user) {
-        analytics.track({ userId: user.sub, event, properties }, resolve);
-      } else {
-        analytics.track({ anonymousId: 'anon', event, properties }, resolve);
-      }
-    });
-  });
-}
+export async function trackUserEvent(event: TrackingEventBase<any>) {
+  console.log('abc');
 
-// make me spawn a process so main thread can end
-export function trackAndSpawn(event: string, properties: any = {}) {
-  getUser().then((user) => {
-    runScriptByName(
-      'emit-analytics',
-      token,
-      JSON.stringify({
-        userId: user ? user.sub : 'anon',
-        event,
-        properties: {
-          ...properties,
-          opticVersion,
-        },
-      })
-    );
-  });
+  const daemonState = await ensureDaemonStarted(
+    lockFilePath,
+    Config.apiBaseUrl
+  );
+
+  const cliServerClient = new Client(
+    `http://localhost:${daemonState.port}/api`
+  );
+
+  cliServerClient.postTrackingEvents([event]);
```

Comment:
This is probably fine since it's local but in the future we may wish to queue & batch flush